### PR TITLE
fix(ui): Fixed login/register with non-unique ids

### LINKED ISSUE #17958: Login/Register page has HTML elements with non-unique IDs
## Important Details

> How are you running Sentry?

Running on my machine for development, from master.

## Description

There are two forms, one in each "tab", and they both have fields with the same ID: `id_username` and `id_password`.

Here's the warning on a Chrome devtools console:

image

## Steps to Reproduce

1. You need Sentry with user registration enabled
2. Visit http://localhost:8000/auth/login/sentry/ (or any other domain)
3. See the warning on the console

### What you expected to happen

There should be no warning. IDs should be unique.

### Possible Solution

The form field `name`s can be retained, and the `id`s updated to `id_registration_username`, `id_registration_password`. Update a JS snippet that queries `#register #id_password` to query `#id_registration_password` instead.

That's all in `src/sentry/templates/sentry/organization-login.html`

### PR DESCRIPTION
**Pull request by priscilawebdev**
closes https://github.com/getsentry/sentry/issues/17958

### CODE REVIEW

File: src/sentry/web/forms/accounts.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ class PasswordlessRegistrationForm(forms.ModelForm):
     username = AllowedEmailField(
         label=_("Email"),
         max_length=128,
-        widget=forms.TextInput(attrs={"placeholder": "you@example.com"}),
+        widget=forms.TextInput(attrs={"placeholder": "you@example.com", "id": "registration-username"}),
```
Comment (rhcarvalho) [inline src/sentry/web/forms/accounts.py]: Long time I don't touch Django code :)
I had to read the docs.

## attrs.id 👍 

Setting the [`id`](https://docs.djangoproject.com/en/1.11/ref/forms/api/#django.forms.BoundField.id_for_label) here is fine, but feels brittle. We had to update 2 fields in 2 different classes, and we have to keep in mind what other fields appear in the other form that is rendered on the same page...

## prefix 👎 

An alternative to consider is to use a [`prefix`](https://docs.djangoproject.com/en/1.11/ref/forms/api/#prefixes-for-forms) when rendering the form or at its definition.

The problem with applying a `prefix` is that it would also change the form field names... and that could break registrations, so better not touch for now. We'd need to guarantee no code depends on the name directly instead of using the Django Forms API (e.g. that piece of JavaScript we're updating didn't use the Forms API).

## auto_id 👍 👍 

The other contender is to use [`auto_id`](https://docs.djangoproject.com/en/1.11/ref/forms/api/#django.forms.Form.auto_id) to let Django set all field IDs consistently.

I suggest uniformly changing all ids in this form by setting a custom `auto_id`:

```python
# src/sentry/web/frontend/auth_login.py

    def get_register_form(self, request, initial=None):
        op = request.POST.get("op")
        return RegistrationForm(
            request.POST if op == "register" else None,
            initial=initial,
            # Custom auto_id to avoid ID collision with AuthenticationForm.
            auto_id="id_registration_%s",
        )
```

Then we don't need to set ids individually:

```suggestion
        widget=forms.TextInput(attrs={"placeholder": "you@example.com"}),
```

File: src/sentry/web/forms/accounts.py [hunk 2]
```diff
@@ -226,7 +226,7 @@ def save(self, commit=True):
 
 class RegistrationForm(PasswordlessRegistrationForm):
     password = forms.CharField(
-        required=True, widget=forms.PasswordInput(attrs={"placeholder": "something super secret"})
+        required=True, widget=forms.PasswordInput(attrs={"placeholder": "something super secret", "id": "registration-password"})
```
Comment (rhcarvalho) [inline src/sentry/web/forms/accounts.py]: If we use `auto_id`, we can revert this change:

```suggestion
        required=True, widget=forms.PasswordInput(attrs={"placeholder": "something super secret"})
```

**Review by rhcarvalho — COMMENTED** [review]
This looks fine, but we could make maintenance easier with a couple of tweaks.

File: src/sentry/templates/sentry/organization-login.html [hunk 3]
```diff
@@ -11,7 +11,7 @@
   SentryApp.passwordStrength.load(function(passwordStrength) {
     $(function() {
       passwordStrength.attachTo({
-        input: document.querySelector('#register #id_password'),
+        input: document.querySelector('#register #registration-password'),
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply src/sentry/templates/sentry/organization-login.html]: I find it a little redundant, since `#` already signals that it is an id ... but I will follow the pattern 😉

File: src/sentry/web/forms/accounts.py (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/web/forms/accounts.py]: super! Thank you for explaining all the possible options ... I didn't know about it

**Review by rhcarvalho — APPROVED** [review]

**Review by markstory — APPROVED** [review]
